### ISSUE TITLE
clickhouse-format formatting issues (minor)

### ISSUE THREAD

**Issue by murfel** [ClickHouse/ClickHouse#48056]
`clickhouse-format --oneline --query "ALTER TABLE alter_test COMMENT COLUMN IF EXISTS ToDrop 'new comment'"`

should output one line:

`ALTER TABLE alter_test COMMENT COLUMN IF EXISTS ToDrop 'new comment'`

but outputs two lines:

```
ALTER TABLE alter_test
    COMMENT COLUMN IF EXISTS ToDrop 'new comment'
```

The issue is found when testing the #45649 refactoring.

I will be fixing the issue in a separate PR.

**Comment by murfel** [comment]
```
$ git diff
diff --git a/src/Parsers/ASTExpressionList.cpp b/src/Parsers/ASTExpressionList.cpp
index 2590c6b294..ddf53aa34c 100644
--- a/src/Parsers/ASTExpressionList.cpp
+++ b/src/Parsers/ASTExpressionList.cpp
@@ -40,7 +40,8 @@ void ASTExpressionList::formatImpl(const FormatSettings & settings, FormatState
 
 void ASTExpressionList::formatImplMultiline(const FormatSettings & settings, FormatState & state, FormatStateStacked frame) const
 {
-    std::string indent_str = "\n" + std::string(4 * (frame.indent + 1), ' ');
+    std::string nl_or_ws = settings.one_line ? " " : "\n";
+    std::string indent_str = settings.one_line ? "" : std::string(4 * (frame.indent + 1), ' ');
 
     if (frame.expression_list_prepend_whitespace)
     {
@@ -59,7 +60,10 @@ void ASTExpressionList::formatImplMultiline(const FormatSettings & settings, For
         }
 
         if (children.size() > 1 || frame.expression_list_always_start_on_new_line)
+        {
+            settings.ostr << nl_or_ws;
             settings.ostr << indent_str;
+        }
 
         FormatStateStacked frame_nested = frame;
         frame_nested.expression_list_always_start_on_new_line = false;
```

**Comment by murfel** [comment]
Missing space before (

```
$ ./programs/clickhouse-format --oneline --hilite --query "create table t as (select 1) comment 'hello'"
CREATE TABLE t AS( SELECT 1) COMMENT 'hello'
$ ./programs/clickhouse-format --hilite --query "create table t as (select 1) comment 'hello'"
CREATE TABLE t AS(
SELECT 1)
COMMENT 'hello'
```

**Comment by murfel** [comment]
The latter issue (misplaced `(`) will be fixed in #47610.
